src/core/Akka/Util/SpanHacks.cs

```diff
@@ -28,7 +26,101 @@ public static int Parse(ReadOnlySpan<char> str)
+
+        /// <summary>
+        /// Can replace with int64.TryFormat in later versions of .NET.
+        /// </summary>
+        /// <param name="i">The integer we want to format into a string.</param>
+        /// <param name="startPos">Starting position in the destination span we're going to write from</param>
+        /// <param name="span">The span we're going to write our characters into.</param>
+        /// <param name="sizeHint">Optional size hint, in order to avoid recalculating it.</param>
+        /// <returns></returns>
+        public static int TryFormat(long i, int startPos, ref Span<char> span, int sizeHint = 0)
+        {
+            var index = 0;
+            if (i is < 10 and >= 0)
+            {
+                span[startPos] = (char)(i+'0');
+                return 1;
+            }
+            
+            var negative = 0;
+            if (i < 0)
+            {
+                negative = 1;
+                i = Math.Abs(i);
+            }
+	
+            var targetLength = sizeHint > 0 ? sizeHint : PositiveInt64SizeInCharacters(i, negative);
+
+            while (i > 0)
+            {
+                i = Math.DivRem(i, 10, out var rem);
+                span[startPos + targetLength - index++ - 1] = (char)(rem+'0');
+            }
+	
+            if(negative == 1){
+                span[0] = Negative;
+                index++;
+            }
+
+            return index;
+        }
+
+        /// <summary>
+        /// How many characters do we need to represent this int as a string?
+        /// </summary>
+        /// <param name="i">The int.</param>
+        /// <returns>Character length.</returns>
+        public static int Int64SizeInCharacters(long i)
+        {
+	        // account for negative characters
+            var padding = 0;
+            if (i < 0)
+            {
+                i *= -1;
+                padding = 1;
+            }
+
+            return PositiveInt64SizeInCharacters(i, padding);
+        }
+        
+        public static int PositiveInt64SizeInCharacters(long i, int padding)
+        {
+            switch (i)
+            {
+                case 0:
+                    return 1;
+                case < 10:
+                    return 1 + padding;
+                case < 100:
+                    return 2 + padding;
+                case < 1000:
+                    return 3 + padding;
+                case < 10000:
+                    return 4 + padding;
+                case < 100000:
+                    return 5 + padding;
+                case < 1_000_000:
+                    return 6 + padding;
+                case < 10_000_000:
+                    return 7 + padding;
+                case < 100_000_000:
+                    return 8 + padding;
+                case < 1_000_000_000:
+                    return 9 + padding;
+                case < 10_000_000_000:
+                    return 10 + padding;
+                case 100_000_000_000:
```

Comment:
Shouldn't this be `case < 100_000_000_000:` ?

Reply:
lol, yep it should. I'll fix that.